widgetastic_manageiq/__init__.py

```diff
@@ -825,6 +825,115 @@ def workaround_save_issue(self):
             "{}._handlers.change.map(function(handler) {{ handler() }});".format(self.item_name))
 
 
+class SettingsGroupSubmenu(NavDropdown):
+    """
+    Widget for the Group 'Dropdown' menu that allows a user assigned to multiple groups
+    to change their active group dynamically Multiple group functionality added in 5.9
+    """
+    GROUP_SUBMENU = './/ul[contains(@class, "dropdown-menu scrollable-menu")]'
+
+    def __init__(self, parent, locator, logger=None):
+        Widget.__init__(self, parent, logger=logger)
+        self.locator = locator
+
+    def __locator__(self):
+        return self.locator
+
+    @property
+    def is_displayed(self):
+        return self.parent.expanded
+
+    @property
+    def expandable(self):
+        try:
+            self.browser.element('./.[contains(@class, "dropdown-submenu")]', parent=self)
+        except NoSuchElementException:
+            return False
+        else:
+            return True
+
+    @property
+    def expanded(self):
+        if not self.expandable:
+            return False
+        return self.browser.element(self.GROUP_SUBMENU, parent=self).is_displayed()
+
+    def expand(self):
+        """ Expand the Settings->Change Group submenu by hovering. """
+        if not self.expandable:
+            raise ValueError('{} is not expandable'.format(self.locator))
+
+        # Make sure the Settings menu is expanded
+        if not self.parent.expanded:
+            self.parent.expand()
+
+        if not self.expanded:
```

Comment:
This check feels unnecessary since its also checked after `self.move_to()` - if hovering while its already expanded doesn't close, I'd recommend skipping this first check and doing the `self.move_to()` in all cases, then checking expanded status once.

Reply:
Good catch.